k8sclient/k8sclient.go

```diff
@@ -69,6 +79,54 @@ func createK8sClient(kubeconfig string) (KubeClient, error) {
 	return &defaultKubeClient{client: client}, nil
 }
 
+func setKubeClientInfo(c *ClientInfo, client KubeClient, k8sArgs types.K8sArgs) {
+	c.Client = client
+	c.Podnamespace = string(k8sArgs.K8S_POD_NAMESPACE)
+	c.Podname = string(k8sArgs.K8S_POD_NAME)
+}
+
+func SetNetworkStatus(k *ClientInfo, netStatus []*types.NetworkStatus) error {
+
+	pod, err := k.Client.GetPod(k.Podnamespace, k.Podname)
+	if err != nil {
+		return fmt.Errorf("SetNetworkStatus: failed to query the pod %v in out of cluster comm: %v", k.Podname, err)
+	}
+
+	var networkStatus []string
+	for _, nets := range netStatus {
+		data, err := json.MarshalIndent(nets, "", "    ")
+		if err != nil {
+			return fmt.Errorf("SetNetworkStatus: error with Marshal Indent: %v", err)
+		}
+		networkStatus = append(networkStatus, string(data))
+	}
+
+	ns := fmt.Sprintf("[%s]", strings.Join(networkStatus, ","))
+
+	_, err = setPodNetworkAnnotation(k.Client, k.Podnamespace, pod, ns)
+	if err != nil {
+		return fmt.Errorf("SetNetworkStatus: failed to update the pod %v in out of cluster comm: %v", k.Podname, err)
+	}
+
+	return nil
+}
+
+func setPodNetworkAnnotation(client KubeClient, namespace string, pod *v1.Pod, ns string) (*v1.Pod, error) {
+	//pod annotations is empty
```

Comment:
Could you add a bit detail like "allocate allocation if pod annotation is empty"?

Reply:
ok cool, Thanks for the suggest, I will make the changes.